action.yml

```diff
@@ -4,7 +4,21 @@ description: 'Authenticate to Azure and run your Az CLI or Az
 inputs: 
   creds:
     description: 'Paste output of `az ad sp create-for-rbac` as value of secret variable: AZURE_CREDENTIALS'
-    required: true
+    required: false

```

Comment:
Should there instead be an input called 'credentialprovider' or 'loginprovider' which takes values like spnwithsecret, spnwithcertificate, and managedidentity and then the action performs login based on this input value. @N-Usha thoughts ?